internal/cmd/controller/finalize/finalize.go

```diff
@@ -96,6 +98,47 @@ func PurgeBundleDeployments(ctx context.Context, c client.Client, bundle types.N
 		if err != nil {
 			return err
 		}
+
+		if err = PurgeContent(ctx, c, bd.Name, bd.Spec.DeploymentID); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// PurgeContent tries to delete the content resource related with the given bundle deployment.
+func PurgeContent(ctx context.Context, c client.Client, name, deplID string) error {
+	contentID, _ := kv.Split(deplID, ":")
+	content := &v1alpha1.Content{}
+	if err := c.Get(ctx, types.NamespacedName{Name: contentID}, content); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	nn := types.NamespacedName{Name: content.Name}
+	if controllerutil.ContainsFinalizer(content, name) {
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			if err := c.Get(ctx, nn, content); err != nil {
+				return err
+			}
+
+			logrus.Infof("Removing finalizer %s from content resource %s", name, contentID)
+			controllerutil.RemoveFinalizer(content, name)
+
+			return c.Update(ctx, content)
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	hasOtherFinalizers := (len(content.Finalizers) > 1) ||
+		((len(content.Finalizers) == 1) && content.Finalizers[0] != name)
+	if !hasOtherFinalizers {
+		logrus.Infof("Deleting content resource %s", contentID)
+		if err := c.Delete(ctx, content); err != nil {
+			return err
+		}
```

Comment:
Question: why checking exactly for 1 finalizer? What is the other finalizer that the Content resource has? 
Should not be deleted when number of finalizers reaches 0?

Reply:
An alternative approach to this could be to eagerly mark a Content resource for deletion, as soon as it is referenced by at least one bundle deployment even when we therefore know that deleting it doesn't make sense. _That_ would allow it to be transparently deleted as soon as its finalizers count reaches 0.